### ISSUE TITLE
Vulnerable package has score 10/10 in Vulnerabilities

### ISSUE THREAD

**Issue by jorgsowa** [ossf/scorecard#3946]
**Describe the bug**
Package https://github.com/elijaa/phpmemcachedadmin has disclosed a vulnerability https://osv.dev/vulnerability/CVE-2023-6026

However, the OSV scanner doesn't detect it, because it scans dependencies, not the package itself (!). As a result, the scorecard gives it 10/10 points.

**Reproduction steps**
1. Run `scorecard --repo=github.com/elijaa/phpmemcachedadmin`
2. See `10 / 10 | Vulnerabilities        | no vulnerabilities detected`

**Expected behavior**
As the vulnerability is known, the score shouldn't be 10/10.

**Additional context**

**Comment by spencerschrock** [comment]
Yeah, I agree we aren't finding vulns in the current project. (See related comment https://github.com/google/osv-scanner/issues/416#issuecomment-1955438494)

I was curious if we did anything differently before the use of osv-scanner (#2509), and it seems like the answer is no
> Currently the vulnerability check only checks if the HEAD commit hash has any vulnerability specified in OSV.dev

> The commit hash check is still being done

**Comment by royy92** [comment]
I reproduced this against the current Scorecard implementation. Running Scorecard against `github.com/elijaa/phpmemcachedadmin` still reports:

```text
10 / 10 | Vulnerabilities | 0 existing vulnerabilities detected
```

I traced the current Vulnerabilities check to verify whether the project itself is being omitted from the scan. Scorecard passes both the repository commit and local repository path to OSV Scanner. In OSV Scanner `v2.3.2`, the supplied commit is ultimately sent to OSV as a commit query.

I verified the behavior directly against the OSV API. For the commit tagged as `1.3.0`:

```text
POST /v1/query
{"commit":"f3dd1e30a84563673a3f1b8237e64e429617a3b1"}
```

OSV returns `CVE-2023-6026` and `CVE-2023-6027`. For the repository HEAD I tested (`aaf3d16a2a1e3dff72dd358f1758d96f0f174cb3`):

```text
POST /v1/query
{"commit":"aaf3d16a2a1e3dff72dd358f1758d96f0f174cb3"}
```

OSV returns no vulnerabilities. This matches the current OSV record for `CVE-2023-6026`. Its generated Git range has the same commit for both boundaries:

```text
introduced:    f3dd1e30a84563673a3f1b8237e64e429617a3b1
last_affected: f3dd1e30a84563673a3f1b8237e64e429617a3b1
```

Because both boundaries resolve to the same commit, the generated Git range is limited to the commit tagged as `1.3.0`.

I also traced the source of this range. The source vulnerability data identifies `1.3.0` as affected without broader version bounds. OSV's CVE conversion represents that as a standalone version and resolves both boundaries to the `1.3.0` tag commit.

As a sanity check, I compared the relevant source between `1.3.0` and the tested HEAD. I did not find a relevant fix in that range, and the relevant file-path handling/deletion path appears unchanged. I have not independently demonstrated exploitability of the tested HEAD, so I don't think this comparison alone is enough to claim that it is definitely affected.

Based on this, I couldn't find a Scorecard-side bug in the current implementation. The project commit is being checked, but OSV itself does not report `CVE-2023-6026` for the tested HEAD. The `10/10` therefore follows the result returned by OSV for that commit rather than Scorecard omitting the project from the scan.

If commits after `1.3.0` are also affected, the affected-version data appears to need clarification or correction upstream. Would you prefer to treat this as an upstream data question, or keep the issue open while the affected range is clarified?
